### ISSUE TITLE
Use after free error detected by Clang trunk

### ISSUE THREAD

**Issue by Xazax-hun** [OpenPrinting/cups-filters#150]
The newest version of clang emits the following warning:
```
filter/rastertopdf.cpp:1301:12: warning: address of stack memory associated with local variable 'iccProfile' returned [-Wreturn-stack-address]
    return iccProfile.c_str();
```
The corresponding code:
https://github.com/OpenPrinting/cups-filters/blob/a71bbd14716ea8cfcb467e0919e12a99375064a6/filter/rastertopdf.cpp#L1301

The method `c_str` will return a pointer that points to the buffer of the std::string object which will be deleted once the std::string is destroyed. Since that object is a local variable, the function will return a dangling pointer.

**Comment by tillkamppeter** [comment]
Gcc does not complain at all, so clang seems to be more intelligent, but it is also not perfect, as both gcc and clang seem to do not issue a warning due to the fact that the getIPPColorProfileName() is nowhere called in rastertopdf. rastertopdf.cxx also compiles without warnings (with gcc at least) if one completely comments out getIPPColorProfileName(),Therefore we probably never got a crash.

**Comment by tillkamppeter** [comment]
Fixed in commit ad64cf9115. Thank you for the bug report.
